stdlib/public/core/ArrayShared.swift

```diff
@@ -222,17 +198,20 @@ extension _ArrayBufferProtocol {
   /// copying when it isUniquelyReferenced.
   @inline(never)
   @inlinable // @specializable
-  internal mutating func _arrayOutOfPlaceUpdate<Initializer>(
+  internal mutating func _arrayOutOfPlaceUpdate(
     _ dest: inout _ContiguousArrayBuffer<Element>,
     _ headCount: Int, // Count of initial source elements to copy/move
     _ newCount: Int,  // Number of new elements to insert
-    _ initializeNewElements: Initializer
-  ) where
-    Initializer: _PointerFunction,
```

Comment:
Idea for an alternative design: replace the closure parameter by returning the newly opened uninitialized gap as an `UnsafeMutableBufferPointer`, which the callee would be expected to fill in.

Reply:
Interesting idea! I think I lean more towards keeping the closure, I like scoping the pointer mutation to as small a scope as possible.